tools/layers.py

```diff
@@ -95,3 +111,37 @@ def set_temporal_settings(
     tprops.setFixedDuration(time_step)
     tprops.setDurationUnits(unit)
     tprops.setIsActive(True)
+
+
+def evaluate_expressions(
+    exp: QgsExpression,
+    feature: Optional[QgsFeature] = None,
+    layer: Optional[QgsMapLayer] = None,
+    context_scopes: Optional[List[QgsExpressionContextScope]] = None,
+) -> Union[bool, int, str, float, None]:
+    """
+    Evaluate a QGIS expression
+    :param exp: QGIS expression
+    :param feature: Optional QgsFeature
+    :param layer: Optional QgsMapLayer
+    :param context_scopes: Optional list of QgsExpressionContextScopes
+    :return: evaluated value of the expression
+    """
+    context = QgsExpressionContext()
+    scopes = context_scopes if context_scopes else []
```

Comment:
Usually explicit check `scopes = context_scopes if context_scopes is not None else default_value` is better so theoretically parameter could be given values 0 or empty strings or something else that evaluates to False.
```sh
$ python -c "import this" | awk 'NR==1||NR==3'
```

Reply:
Yeah, usually I just check if it is None, but today I was lazy for some reason :D